### PR TITLE
Adding basic visualization for organization list using P5.js

### PR DESCRIPTION
**Pull request by blinfoldking**

### CODE REVIEW

File: jsonloader.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+loadJSON = (callback) => {
+
+    var xobj = new XMLHttpRequest();
+    xobj.overrideMimeType("application/json");
+    xobj.open('GET', './_data/orgs.json', true);
```
Comment (jayvdb) [inline jsonloader.js]: this file wasnt being published, but I've fixed that: https://github.com/summerofcode/gsoc-orgs/commit/34243808e3d2afabd397c435756767bafdaf2d72

**Review by jayvdb — CHANGES_REQUESTED** [review]
Please rebase, and also use CDN URLs instead of adding vendor javascript code to this repo

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by jayvdb** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by jayvdb** [comment]
yay, preview is working https://deploy-preview-16--soc-orgs.netlify.com/list.html

File: visualizer.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+      }
+      break
+    case ORGS_MODE:
+	let changeState = true
+	for(let index = 0; index < datPos.length; index++){
+	    if(dist(...mousePos, ...datPos[index]) <= 10){
+
+        document.getElementById('title').innerHTML = node[index].name
+        document.getElementById('desc').innerHTML = node[index].tagline
+        document.getElementById('web').setAttribute('href', node[index].website_url
```
Comment (jayvdb) [inline visualizer.js]: change this to point to the generated page, e.g. https://soc-orgs.netlify.com/orgs/aimacode

File: jsonloader.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    xobj.onreadystatechange = () => {
+        if (xobj.readyState == 4 && xobj.status == "200") {
```
Comment (wisn) [inline jsonloader.js]: Use `===` instead of `==`. `"200"` should be an integer.

File: list.html [hunk 4]
```diff
@@ -6,4 +7,72 @@
...
+
+</url>-->
+<style>
+  body{
+    margin: 0;
+  }
+
+
+  #container {
+		width: 25%;
```
Comment (wisn) [inline list.html]: Inconsistent indentation.

File: jsonloader.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            callback(xobj.responseText);
+
+        }
+    }
+    xobj.send(null);
+
+}
+
+
+loadData = (cb) => {
```
Comment (wisn) [inline jsonloader.js]: `(cb)` doesn't need brackets. `loadData = cb => {` is fine.

File: visualizer.js [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+    centrex + random(-radius, radius) * tolerance,
+    centrey + random(-radius, radius) * tolerance 
+  ]
+}
+
+
+loadData(json => {
+  json.forEach(element => {
+    data.push(element)
+    if(category.find(value => value.category_name == element.category) == null)
```
Comment (wisn) [inline visualizer.js]: Give a space after `if` keyword and use `===` instead of `==`.

File: visualizer.js [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+let selectedCategory = category[0]
+let node = []
+
+const view = [
+  screenWidth()/2, 
+  screenHeight()/2
+]
+
+const getcoord = (x, y) => {
+  return [x - screenWidth()/2, y - screenHeight()/2]
```
Comment (wisn) [inline visualizer.js]: `const getcoord = (x, y) => [x - screenWidth()/2, y - screenHeight()/2]` is equivalence.

**Review by wisn — CHANGES_REQUESTED** [review]
We need to use a JavaScript standard style such as [Standard](https://standardjs.com/).

File: visualizer.js [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+      }
+      break
+    case ORGS_MODE:
+      let changeState = true
+      for (let index = 0; index < datPos.length; index++) {
+        if (dist(...mousePos, ...datPos[index]) <= 10) {
+
+          document.getElementById('title').innerHTML = node[index].name
+          document.getElementById('desc').innerHTML = node[index].tagline
+          document.getElementById('web').setAttribute('href', node[index].website_url)
```
Comment (jayvdb) [inline visualizer.js]: This still needs fixing: https://github.com/summerofcode/gsoc-orgs/pull/16#discussion_r173640150

**Review by jayvdb — CHANGES_REQUESTED** [review]

**Comment by jayvdb** [comment]
On the view of a single category, with lots of orgs, what is determining the layout.  The placement of the org nodes doesnt have an obvious purpose.  If there is a good reason for the layout, maybe add a note explaining it to the user.

**Comment by BlinfoldKing** [comment]
the orgs is currently spawn at random position, i'm currently still searching for a way to represent the data.

**Review by jayvdb — CHANGES_REQUESTED** [review]
Please use CDN URLs instead of adding vendor javascript code to this repo

**Comment by jayvdb** [comment]
Also need to replace `jquery` & `p5` with CDN versions.

**Comment by BlinfoldKing** [comment]
i've already replace it with the cdn version

**Comment by jayvdb** [comment]
Nice, I must have been looking at an old version.

Can you lint all of your JS code with [`standard`](https://github.com/standard/standard).  c.f. https://github.com/summerofcode/gsoc-orgs/issues/19

Then squash your commits into a single commit if you want to.  If you dont, I have to squash it, which means that I become the git `Committer` on your patch.

**Pull request merged**